libs/@local/harpc/crates/wire/src/encoding.rs

```diff
@@ -0,0 +1,147 @@
+use std::io;
+
+use enumflags2::BitFlags;
+use error_stack::{Report, Result, ResultExt};
+use tokio::io::{AsyncRead, AsyncWrite};
+
+use crate::codec::{DecodePure, Encode};
+
+// TODO: this might be problematic in the future, as it only allows for 16 different encodings.
+// we could also use a more packed encoding, in that case `Accept` would need to not be `BitFlag`
+// but instead a `ItemOrVec` that's encoded, which is a bit more messy.
+#[enumflags2::bitflags]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(test, derive(test_strategy::Arbitrary))]
+#[repr(u16)]
+pub enum Encoding {
+    Raw = 1 << 0,
+    Json = 1 << 1,
+    Cbor = 1 << 2,
+}
```

Comment:
This is because of the `accept` header to address a concern that was raised back in January. The attempt here is to simulate the `Content-Type` and `Accept` header, so that you can be sure that the encoding you get is the one you expected.

I am leaning towards removing this entirely and just say: we know what the encoding will be

Reply:
Let's remove it 🚀

Reply:
IIRC in January we decided to start with JSON as encoding, but I don't think this has to be specified, this would just be an implementation detail

Reply:
removed